merkle-tree/hash-set/src/lib.rs

```diff
@@ -0,0 +1,1211 @@
+        hs.insert(&element_1_1, 0).unwrap();
+        hs.mark_with_sequence_number(&element_1_1, 1).unwrap();
+
+        // Check if element exists in the set.
+        assert_eq!(hs.contains(&element_1_1, 1).unwrap(), true);
+        // Try inserting the same element, even though we didn't reach the
+        // threshold.
+        assert!(matches!(
+            hs.insert(&element_1_1, 1),
+            Err(HashSetError::ElementAlreadyExists)
+        ));
+
+        // Insert multiple elements and mark them with one sequence number.
+        // An equivalent to a batched insertion in Light Protocol.
+
+        let element_2_3 = 3.to_biguint().unwrap();
+        let element_2_6 = 6.to_biguint().unwrap();
+        let element_2_8 = 8.to_biguint().unwrap();
+        let element_2_9 = 9.to_biguint().unwrap();
+        hs.insert(&element_2_3, 1).unwrap();
+        hs.insert(&element_2_6, 1).unwrap();
+        hs.insert(&element_2_8, 1).unwrap();
+        hs.insert(&element_2_9, 1).unwrap();
+        assert_eq!(hs.contains(&element_2_3, 2).unwrap(), true);
+        assert_eq!(hs.contains(&element_2_6, 2).unwrap(), true);
+        assert_eq!(hs.contains(&element_2_8, 2).unwrap(), true);
+        assert_eq!(hs.contains(&element_2_9, 2).unwrap(), true);
+        hs.mark_with_sequence_number(&element_2_3, 2).unwrap();
+        hs.mark_with_sequence_number(&element_2_6, 2).unwrap();
+        hs.mark_with_sequence_number(&element_2_8, 2).unwrap();
+        hs.mark_with_sequence_number(&element_2_9, 2).unwrap();
+        assert!(matches!(
+            hs.insert(&element_2_3, 2),
+            Err(HashSetError::ElementAlreadyExists)
+        ));
+        assert!(matches!(
+            hs.insert(&element_2_6, 2),
+            Err(HashSetError::ElementAlreadyExists)
+        ));
+        assert!(matches!(
+            hs.insert(&element_2_8, 2),
+            Err(HashSetError::ElementAlreadyExists)
+        ));
+        assert!(matches!(
+            hs.insert(&element_2_9, 2),
+            Err(HashSetError::ElementAlreadyExists)
+        ));
+
+        let element_3_11 = 11.to_biguint().unwrap();
+        let element_3_13 = 13.to_biguint().unwrap();
+        let element_3_21 = 21.to_biguint().unwrap();
+        let element_3_29 = 29.to_biguint().unwrap();
+        hs.insert(&element_3_11, 2).unwrap();
+        hs.insert(&element_3_13, 2).unwrap();
+        hs.insert(&element_3_21, 2).unwrap();
+        hs.insert(&element_3_29, 2).unwrap();
+        assert_eq!(hs.contains(&element_3_11, 3).unwrap(), true);
+        assert_eq!(hs.contains(&element_3_13, 3).unwrap(), true);
+        assert_eq!(hs.contains(&element_3_21, 3).unwrap(), true);
+        assert_eq!(hs.contains(&element_3_29, 3).unwrap(), true);
+        hs.mark_with_sequence_number(&element_3_11, 3).unwrap();
+        hs.mark_with_sequence_number(&element_3_13, 3).unwrap();
+        hs.mark_with_sequence_number(&element_3_21, 3).unwrap();
+        hs.mark_with_sequence_number(&element_3_29, 3).unwrap();
```

Comment:
```suggestion
        hs.mark_with_sequence_number(&element_3_11, 4).unwrap();
        hs.mark_with_sequence_number(&element_3_13, 4).unwrap();
        hs.mark_with_sequence_number(&element_3_21, 4).unwrap();
        hs.mark_with_sequence_number(&element_3_29, 4).unwrap();
```

Reply:
I don't understand why. I think the sequence numbers are fine as they are?

Reply:
resolved with your explanation above